src/main/c/powerwaf_jni.c

```diff
@@ -527,6 +538,17 @@ static jobject _run_rule_common(bool is_byte_buffer, JNIEnv *env, jclass clazz,
             JNI(DeleteLocalRef, thr);
             goto end;
         }
+
+        ephemeral_input = _convert_checked(env, ephemeral_data, &limits, 0);
+        thr = JNI(ExceptionOccurred);
+        if (thr) {
+            JAVA_LOG_THR(DDWAF_LOG_INFO, thr,
+                         "Exception encoding parameters");
+            java_wrap_exc("%s", "Exception encoding parameters");
+            JNI(DeleteLocalRef, thr);
+            goto end;
```

Comment:
This is leaking `input`. Additionally, the current code is also leaking  `input` if `_get_time_checked` fails (should never happen) or `get_gen_budget_in_us` returns 0.

I think the best here would be to destroy all the inputs on the `end` label if we didn't go the bytebuffer path. But also, if there is a call do `ddwaf_run` that doesn't result in `DDWAF_ERR_INVALID_ARGUMENT`, then the inputs should be reassigned `_pwinput_invalid` to avoid `ddwaf_object_free` have any effect  because then data is to be destroyed when the additive is destroyed (i.e. the data is then owned by the additive).

Reply:
Fixed